src/azure/accountStore.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import vscode = require('vscode');
+import { IAccount } from '../models/contracts/azure/accountInterfaces';
+import Constants = require('../constants/constants');
+import AzureAuth = require('@cssuh/ads-adal-library');
+
+export class AccountStore {
+    constructor(
+        private _context: vscode.ExtensionContext
+    ) { }
+
+    public getAccounts(): IAccount[] {
+        let configValues = this._context.globalState.get<IAccount[]>(Constants.configAzureAccount);
+        if (!configValues) {
+            configValues = [];
+        }
+        return configValues;
+    }
+
+    public getAccount(key: string): IAccount {
+        let account: IAccount;
+        let configValues = this._context.globalState.get<IAccount[]>(Constants.configAzureAccount);
+        if (!configValues) {
+            // Throw error message saying there are no accounts stored
+        }
+        for (let value of configValues) {
+            if (value.key.id === key) {
+                account = value;
+                break;
+            }
+        }
+        if (!account) {
+            // Throw error message saying the account was not found
+            return undefined;
+        }
+        return account;
+    }
+
+    public removeAccount(key: string): void {
+        return;
+    }
+
+    /**
+     * Adds an account to the account store.
+     *
+     * @param {IAccount} account the account to add
+     * @returns {Promise<void>} a Promise that returns when the account was saved
+     */
+    public addAccount(account: IAccount): Promise<void> {
+        const self = this;
```

Comment:
Don't do this

Justification: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-this-alias.md